FWCore/Framework/interface/EventSetupProvider.h

```diff
@@ -111,32 +110,33 @@ namespace edm {
 
       /// Intended for use only in tests
       void addRecord(const EventSetupRecordKey& iKey);
+      void setPreferredProviderInfo(PreferredProviderInfo const& iInfo);
 
-    protected:
-      void insert(std::unique_ptr<EventSetupRecordProvider> iRecordProvider);
+      void fillRecordsNotAllowingConcurrentIOVs(std::set<EventSetupRecordKey>& recordsNotAllowingConcurrentIOVs) const;
 
-    private:
-      EventSetupProvider(EventSetupProvider const&) = delete;  // stop default
+      void fillKeys(std::set<EventSetupRecordKey>& keys) const;
 
-      EventSetupProvider const& operator=(EventSetupProvider const&) = delete;  // stop default
+      EventSetupRecordProvider* tryToGetRecordProvider(const EventSetupRecordKey& iKey);
```

Comment:
EventSetupsController calls this function when it is creating and initializing the EventSetupRecordIOVQueue objects. The function is public so that EventSetupsController can call it. There is one EventSetupRecordIOVQueue per record. It has a pointer to each EventSetupRecordProvider that corresponds to that record. If there are no SubProcesses, then there is only one, but with SubProcesses there can be more than one and this is what is introducing this and some other complications into this PR.